apidef/oas/operation.go

```diff
@@ -501,27 +502,115 @@ func isRegex(value string) bool {
 func splitPath(inPath string) ([]pathPart, bool) {
 	// Each URL fragment can contain a regex, but the whole
 	// URL isn't just a regex (`/a/.*/foot` => `/a/{param1}/foot`)
-	parts := strings.Split(strings.Trim(inPath, "/"), "/")
+	inPath = strings.Trim(inPath, "/")
+
+	if inPath == "" {
+		return []pathPart{}, false
+	}
+
+	if err := validatePath(inPath); err != nil {
+		return []pathPart{}, false
+	}
+
+	parts := strings.Split(inPath, "/")
+
 	result := make([]pathPart, len(parts))
 	found := 0
+	nCustomRegex := 0
+
+	trimPathParam := func(value string) string {
+		value = strings.TrimPrefix(value, "{")
+		value = strings.TrimSuffix(value, "}")
+
+		return value
+	}
 
 	for k, value := range parts {
-		name := value
-		isRegex := isRegex(value)
-		if isRegex {
+		// Handle non-bracketed path segments
+		// for example: /a/b/c, /a/[0-9]
+		if !strings.HasPrefix(value, "{") && !strings.HasSuffix(value, "}") {
+			name := value
+
+			result[k] = pathPart{
+				name:  name,
+				value: value,
+			}
+
+			if isRegex(value) {
+				nCustomRegex++
+				found++
+
+				result[k].name = fmt.Sprintf("customRegex%d", nCustomRegex)
+				result[k].isRegex = true
+			}
+
+			continue
+		}
+
+		// Handle bracketed path segments
+		segment := trimPathParam(value)
```

Comment:
```suggestion
		segment := strings.Trim(value, "{}")
```

Reply:
Yeah, I don't think anything else is valid, because it would be `{{id}}` or `{{id}` or `{id:...}}` (all invalid).

Reply:
Fixed, PTAL